src/app/dim-api/reducer.ts

```diff
@@ -1203,7 +1203,7 @@ function saveSearch(
   const { canonical, saveable } = parseAndValidateQuery(query, searchConfigs, {
     customStats: draft.settings.customStats ?? [],
   } as FilterContext);
-  if (!saveable) {
+  if (!saveable && saved) {
```

Comment:
Forgive me, I only learnt DIM was open-source this morning when I opened #9268, but shouldn't this be `&& !saved`? If it's saved already you still want to be allowed to unsave it.

Reply:
The argument has a bit of a confusing name -- this is the function responsible for updating the state, and `saved` indicates the desired state. So we shouldn't allow an invalid query it if the desired state is "saved".

Reply:
Amazing, thanks for the explanation